### ISSUE TITLE
FATAL: Failed to extract file jdk-8u172-linux-x64.tar.gz

### ISSUE THREAD

**Issue by edinarjenkins** [sous-chefs/java#478]
### Cookbook version
2.1.0

### Platform Details
Ubuntu

### Scenario:
```
    * apt_package[tar] action install[2018-07-19T16:56:25+00:00] INFO: Processing apt_package[tar] action install (/var/chef/cache/cookbooks/java/resources/oracle_install.rb line 105)
tar:
  Installed: 1.28-2.1ubuntu0.1
  Candidate: 1.28-2.1ubuntu0.1
  Version table:
 *** 1.28-2.1ubuntu0.1 500
        500 http://us-east-2.ec2.archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages
        100 /var/lib/dpkg/status
     1.28-2.1 500
        500 http://us-east-2.ec2.archive.ubuntu.com/ubuntu xenial/main amd64 Packages
 (up to date)

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
[2018-07-19T16:56:25+00:00] FATAL: Failed to extract file jdk-8u172-linux-x64.tar.gz!
[2018-07-19T16:56:25+00:00] INFO: Running queued delayed notifications before re-raising exception

```
### Steps to Reproduce:
run chef-client with default recipe

**Comment by tophercullen** [comment]
I just ran into the same problem. Looks like 172 doesn't exist anymore on oracles site

**Comment by espoelstra** [comment]
This can be worked around by supplying the new version's URL and checksum directly via attributes in your wrapper cookbook. This will work with future versions as well, you just need to manually look up the URL and checksum after a new release drops (and breaks the world).

```
default['java']['jdk']['8']['x86_64']['url'] = 'http://download.oracle.com/otn-pub/java/jdk/8u181-b13/96a7b8442fe848ef90c96a2fad6ed6d1/jdk-8u181-linux-x64.tar.gz'
default['java']['jdk']['8']['x86_64']['checksum'] = '1845567095bfbfebd42ed0d09397939796d05456290fb20a83c476ba09f991d3'
```

It is much safer to copy the JRE/JDK somewhere inside your infrastructure even if it is just an S3 bucket and point to that as opposed to having your processes fail when Oracle unpublishes old (insecure) versions.

It would be amazing if somebody set up some kind of a trigger that when a Java related process fails something scrapes the Oracle site and finds the right version link and checksum and submits a PR here.

**Comment by tophercullen** [comment]
I already use a wrapper cookbook and ran into the same problem. The error handling is what needs to be fixed. It shouldn't even reach the extraction stage if the HTTP request fails.

**Comment by majormoses** [comment]
fixed by #479
